scripts/analysis.py

```diff
@@ -999,43 +967,45 @@ def stacked_bar_chart(dictionary, timestep,
             label = str(key)
         # very first country does not have a 'bottom' argument
         if sum(dictionary[key]) == 0:
-            print(str(dictionary[key]) + 'has no values')
+            print(label + ' has no values')
         elif top_index is True:
-            plot = plt.bar(left=init_year + (timestep/12),
+            plot = plt.bar(left=init_year + (timestep / 12),
                            height=dictionary[key],
                            width=0.1,
-                           color=cm.viridis(1.*color_index/len(dictionary)),
-                           edgecolor='none',
-                           label=label)
+                           color=cm.viridis(
+                1. * color_index / len(dictionary)),
+                edgecolor='none',
+                label=label)
             prev = dictionary[key]
-            ax = plt.gca()
-            ax.get_yaxis().set_major_formatter(plt.FuncFormatter(lambda x, loc: "{:,}".format(int(x))))
             top_index = False
             plot_list.append(plot)
 
         # All curves except the first have a 'bottom'
         # defined by the previous curve
         else:
-            plot = plt.bar(left=init_year + (timestep/12),
+            plot = plt.bar(left=init_year + (timestep / 12),
```

Comment:
if your numpy array isn't initialized with a float dtype, then yes, it can still end up as all integers and cause drama. If you don't believe it, write a test.